File: tests/fixtures/code_search/javascript/simple.js
/**
 * Simple JavaScript file with functions and classes for testing.
 */

function calculateTotal(items) {
  let total = 0;
  for (const item of items) {
    total += item.price;
  }
  return total;
}

const processData = (data) => {
  return data.map(item => item * 2);
};

class ShoppingCart {
  constructor() {
    this.items = [];
  }

  addItem(item) {
    this.items.push(item);
  }

  getTotal() {
    return calculateTotal(this.items);
  }
}

class Product {
  constructor(name, price) {
    this.name = name;
    this.price = price;
  }

  applyDiscount(percentage) {
    this.price *= 1 - (percentage / 100);
  }
}

export { calculateTotal, processData, ShoppingCart, Product };

File: tests/fixtures/code_search/python/simple.py
"""Simple Python file with basic functions and classes for testing."""


def calculate_total(items):
    """Calculate total price of items."""
    total = 0
    for item in items:
        total += item.price
    return total


def process_data(data):
    """Process data and return results."""
    results = []
    for item in data:
        results.append(item * 2)
    return results


class ShoppingCart:
    """Shopping cart class."""

    def __init__(self):
        """Initialize shopping cart."""
        self.items = []

    def add_item(self, item):
        """Add item to cart."""
        self.items.append(item)

    def get_total(self):
        """Get cart total."""
        return calculate_total(self.items)


class Product:
    """Product class."""

    def __init__(self, name, price):
        """Initialize product."""
        self.name = name
        self.price = price

    def apply_discount(self, percentage):
        """Apply discount to product."""
        self.price *= 1 - (percentage / 100)
